src/selectors/resolveNotePath.ts

```diff
@@ -0,0 +1,45 @@
+import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
+import appendToPath from '../util/appendToPath'
+import head from '../util/head'
+import findDescendant from './findDescendant'
+import { anyChild } from './getChildren'
+import getThoughtById from './getThoughtById'
+
+interface NoteKeyResult {
+  noteKey: string
+  noteId?: ThoughtId
+}
+
+/** Resolves note key and note id by checking for note thoughts. */
+export const resolveNoteKey = (state: State, thoughtId: ThoughtId): NoteKeyResult => {
```

Comment:
While the co-location is nice, I want to stick to one default export per file as module as possible since that is a convention on the project.

See: d7fc8936044